File: recursividade/recursividade_palindromo.c
#include <stdio.h>
#include <string.h>

int palindromo(char *text, int n){
    if (n <= 0) return 1;

    if(text[0] != text[n - 1]) return 0;

    return palindromo(text + 1, n - 2);

}

int main(){

    char text[] = "arara";
    int tamanho = strlen(text);

    int resultado = palindromo(text, tamanho);

    if(!resultado) printf("Nao eh palindromo\n");
    else printf("Eh palindromo\n");

    return 0;
}

File: recursividade/recursividade_palindromo.py
def palindromo(text):
    if len(text) <= 1: return 1
    if text[0] != text[-1]: return 0
    return palindromo(text[1:-1])

text = "arara"
resultado = palindromo(text)
if resultado == 0:
    print("Nao eh palindromo")
else:
    print("Eh palindromo")